binary/plan.proto

```diff
@@ -10,18 +10,18 @@ option csharp_namespace = "Substrait.Protobuf";
 
 // Either a relation or root relation
 message PlanRel {
-    oneof RelType {
-        // Any relation
-        Rel rel = 1;
-        // The root of a relation tree
-        RelRoot root = 2;
-    }
+  oneof RelType {
```

Comment:
As part of this patch, can we make sure that we also check for other things:
* require proto3 syntax
* no use of excessive optional, packed references
* names follow standards:
  * snake_case for packages, files, fields, oneof kinds (this one was updated inconsistently and didn't notice until after merge)
  * CamelCase for messages, enum

Reply:
I think we can do that, but I'd rather do it with a tool like [`buf`](https://docs.buf.build/introduction) (they provide [useful github actions](https://docs.buf.build/ci-cd/github-actions) to do this).

I think `prototool` is the only other alternative, and they recommend looking at `buf`!

Reply:
I'd like to do that in a follow up because there are some changes that the buf linter recommends and I don't think those should be bundled in this PR.

Reply:
How much hand work was this PR (versus automatic)? Asking as it may be painful to rebase #80 on top of these changes (I haven't tried, just guessing).

Reply:
All of the work here except the CI bits was automatic, so virtually no hand work was done. Happy to wait for #80 before landing this.

Reply:
`buf lint` is quite nice.  Additionally `buf breaking` is awesome for drift detection